src/__/arrays/flatten.php

```diff
@@ -19,26 +44,20 @@
  * [1, 2, 3, 4]
  * ```
  *
- * @param array $array
- * @param bool $shallow
+ * @since 0.2.0 iterable objects are now supported
  *
- * @return array
+ * @param iterable $iterable
+ * @param bool     $shallow
+ *
+ * @return array|\Generator
  */
-function flatten($array, $shallow = false)
+function flatten($iterable, $shallow = false)
 {
-    $output = [];
-    foreach ($array as $value) {
-        if (is_array($value)) {
-            if (!$shallow) {
-                $value = flatten($value, $shallow);
-            }
-            foreach ($value as $valItem) {
-                $output[] = $valItem;
-            }
-        } else {
-            $output[] = $value;
-        }
+    $generator = flattenIterable($iterable, $shallow);
+
+    if (is_array($iterable)) {
+        return iterator_to_array($generator);
```

Comment:
Like the pattern (common implementation based on iterable + `iterator_to_array`) :+1: